models/Agriculture_price_prediction/app.py

```diff
@@ -0,0 +1,259 @@
+            z-index: 2; /* Ensure text is above overlay */
+        }
+        h3 {
+            color: #333333;
+            text-shadow: 1px 1px 2px rgba(0, 0, 0, 0.2);
+            position: relative;
+            z-index: 2; /* Ensure text is above overlay */
+        }
+        footer {
+            color: #555555;
+            text-align: center;
+            padding: 15px 0;
+            font-size: 14px;
+            border-top: 1px solid #ddd;
+            position: relative;
+            z-index: 2; /* Ensure text is above overlay */
+        }
+        .stats-box {
+            background-color: rgba(255, 255, 255, 0.9); /* Solid background for statistics */
+            padding: 15px;
+            border-radius: 12px;
+            border: 1px solid #ddd;
+            box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
+            margin: 20px 0;
+        }
+    </style>
+    """,
+    unsafe_allow_html=True
+)
+
+# Add an overlay div
+st.markdown('<div class="overlay"></div>', unsafe_allow_html=True)
+
+# Title with gradient and shadow
+st.markdown("<h1 style='text-align: center;'>Agricultural Price Forecasting</h1>", unsafe_allow_html=True)
+st.markdown(
+    "<h3 style='text-align: center;'>Predict and Visualize Commodity Prices Based on Rainfall and Temperature</h3>",
+    unsafe_allow_html=True)
+
+# Sidebar for user inputs with better organization
+st.sidebar.markdown("<h2 style='color: #0056b3;'>User Inputs</h2>", unsafe_allow_html=True)
+
+with st.sidebar.expander("Select Parameters", expanded=True):
+    state = st.selectbox("Select State", ["Mumbai"],
+                         help="Choose the state for which you want to forecast prices.")
+    commodity = st.selectbox("Select Commodity", ["Tur", "Moong", "Urad", "Masur"],
+                             help="Choose the commodity to analyze.")
+    days_to_predict = st.slider("Select number of days to predict", min_value=1, max_value=60, value=30,
+                                help="Select the number of days for price forecasting.")
+    rainfall = st.slider("Enter Rainfall (in inches)", min_value=0, max_value=600, value=25,
+                         help="Enter the rainfall amount in inches.")
+    temperature = st.slider("Enter Temperature (in °C)", min_value=-5, max_value=50, value=20,
+                            help="Enter the temperature in degrees Celsius.")
+
+# File paths
+file_path_delhi = 'Data\\agrodelhhi.xlsx'  # Replace with the path to your Excel file for Delhi
+file_path_mumbai_tur = 'Data\\MumbaiTurExcl.xlsx'  # Replace with the path to your Excel file for Mumbai Tur
+file_path_mumbai_masur = 'Data\\Mumbai_masur.xlsx'  # Replace with the path to your Excel file for Mumbai Masur
+file_path_mumbai_moong = 'Data\\Mumbai_moong.xlsx'  # Replace with the path to your Excel file for Mumbai Moong
+file_path_mumbai_urad = 'Data\\Mumbai_urad_.xlsx'  # Replace with the path to your Excel file for Mumbai Urad
+
+
+# Load data
+@st.cache_data
+def load_data(file_path):
```

Comment:
### 🔴 Step 3:  
Please move this code to a separate file within the existing folder to enhance modularity and maintainability in the codebase.